test/suite/run.py

```diff
@@ -369,6 +396,68 @@ def testsFromArg(tests, loader, arg, scenario):
             sys.exit(2)
         testargs.append(arg)
 
+    if asan:
+        # To run ASAN, we need to ensure these environment variables are set:
+        #    ASAN_SYMBOLIZER_PATH    full path to the llvm-symbolizer program
+        #    LD_LIBRARY_PATH         includes path with wiredtiger shared object
+        #    LD_PRELOAD              includes the ASAN runtime library
+        #
+        # Note that LD_LIBRARY_PATH has already been set above. The trouble with
+        # simply setting these variables in the Python environment is that it's
+        # too late. LD_LIBRARY_PATH is commonly cached by the shared library
+        # loader at program startup, and that's already been done before Python
+        # begins execution. Likewise, any preloading indicated by LD_PRELOAD
+        # has already been done.
+        #
+        # Our solution is to set the variables as appropriate, and then restart
+        # Python with the same argument list. The shared library loader will
+        # have everything it needs on the second go round.
+
+        ASAN_ENV = "__WT_TEST_SUITE_ASAN"    # if set, we've been here before
+        ASAN_SYMBOLIZER_PROG = "llvm-symbolizer"
+        ASAN_SYMBOLIZER_ENV = "ASAN_SYMBOLIZER_PATH"
+        LD_PRELOAD_ENV = "LD_PRELOAD"
+        SO_FILE_NAME = "libclang_rt.asan-x86_64.so"
```

Comment:
Don, are you sure the file name is the same for all the clang versions we support?

Reply:
I did a global search on my dev host and the filename is the same for all the clang versions we have.